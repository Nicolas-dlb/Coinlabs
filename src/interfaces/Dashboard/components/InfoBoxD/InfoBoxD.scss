.infoboxD {
  border-radius: 12px;
  transition: background-color 0.3s ease;
  background-color: var(--third-bg-color);
  width: 46vw;
  height: 8vh;
  display: flex;
  flex-direction: column;
  align-items: space-evenly;
  justify-content: space-between;
  padding: 20px;

  .infobox_top {
    display: flex;
    justify-content: space-between;
    align-items: center;
    color: #c4c3c9;
    .infobox_top_variation {
      display: flex;
      align-items: center;
      justify-content: center;
      svg {
        height: 14px;
        width: 14px;

        margin-right: 5px;
      }
      color: #37bb77;
    }
  }
  .infobox_bottom {
    font-size: 18px;
    width: 100%;
    display: flex;
    p:first-child {
      margin-right: 5px;
    }
    input {
      position: relative;
      top: 1px;
      font-size: 20px;
      cursor: pointer;
      background: var(--third-bg-color);
      outline: none;
      border: none;
      color: var(--primary-font-color);
      width: 95%;
      caret-color: #1baa63;
      &::-webkit-outer-spin-button,
      &::-webkit-inner-spin-button {
        -webkit-appearance: none;
        margin: 0;
      }

      &[type="number"] {
        -moz-appearance: textfield;
      }
    }
  }
}

@media only screen and (min-width: 992px) {
  .infoboxD {
    margin: 0;
    width: 30%;
    border-right: 1px solid #3b374d;
    border-radius: 0;
    padding: 0vh 4.5vw;
    margin-right: 0 !important;
    &:first-child {
      padding-left: 20px;
    }

    .infobox_bottom {
      font-size: 26px;
    }
  }
}
