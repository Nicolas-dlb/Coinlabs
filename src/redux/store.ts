import { configureStore } from "@reduxjs/toolkit";
import marketReducer from "./reducers/marketSlice";
import walletsReducer from "./reducers/walletsSlice";
import appReducer from "./reducers/appSlice";
import userReducer from "./reducers/userSlice";

export default configureStore({
  reducer: {
    market: marketReducer,
    wallets: walletsReducer,
    app: appReducer,
    user: userReducer,
  },
});
