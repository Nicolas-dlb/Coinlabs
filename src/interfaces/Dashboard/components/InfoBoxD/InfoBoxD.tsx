import { useDispatch, useSelector } from "react-redux";
import { selectWallets, setWallets } from "redux/reducers/walletsSlice";
import { getNumberFixed, lastUpdate, numberWithSpaces } from "utils/utils";
import "./InfoBoxD.scss";
import $ from "jquery";
import React, { useEffect } from "react";
import { auth } from "firebaseConfig";

type InfoBoxProps = {
  title: string;
  variation?: number;
  input?: true;
};
function InfoBox({ title, variation, input }: InfoBoxProps) {
  const wallet = useSelector(selectWallets);
  const color = variation && variation > 0 ? "#37bb77" : "#f1c94f";
  const angle = variation && variation > 0 ? "140deg" : "215deg";
  const amount =
    title === "Balance"
      ? wallet.usd
      : title === "Crypto"
      ? wallet.TotalCrypto
      : wallet[title.toLowerCase()];
  const dispatch = useDispatch();

  const setBalance = () => {
    const value = $("#balance").val();
    dispatch(
      setWallets({
        usd: Number(value),
      })
    );
    lastUpdate(dispatch, auth.currentUser?.uid);
    $("#balance").val(getNumberFixed(value, 2));
  };

  useEffect(() => {
    $("#balance").val(getNumberFixed(wallet.usd, 2));
  }, []);
  return (
    <div className="infoboxD">
      <div className="infobox_top">
        <p>{title}</p>
        {variation && (
          <div className="infobox_top_variation">
            <svg
              style={{ transform: `rotate(${angle})` }}
              xmlns="http://www.w3.org/2000/svg"
              viewBox="0 0 492 492"
              fill={color}
            >
              <path
                strokeWidth="2"
                d="M464.3 207.4l0.8 0.2H135.9l103.5-103.7c5.1-5.1 7.8-11.9 7.8-19.1 0-7.2-2.8-14-7.8-19.1L223.3 49.5c-5.1-5.1-11.8-7.9-19-7.9 -7.2 0-14 2.8-19 7.8L7.8 226.9C2.8 232 0 238.8 0 246c0 7.2 2.8 14 7.8 19.1l177.4 177.4c5.1 5.1 11.8 7.8 19 7.8 7.2 0 13.9-2.8 19-7.8l16.1-16.1c5.1-5.1 7.8-11.8 7.8-19 0-7.2-2.8-13.6-7.8-18.7L134.7 284.4h330c14.8 0 27.3-12.8 27.3-27.6v-22.8C492 219.2 479.2 207.4 464.3 207.4z"
              />
            </svg>
            <p style={{ color }}>{variation}%</p>
          </div>
        )}
      </div>
      <div className="infobox_bottom">
        <p>$</p>
        {!input ? (
          <>
            <p>{numberWithSpaces(getNumberFixed(amount, 2))}</p>
          </>
        ) : (
          <>
            <input onBlur={setBalance} id="balance" type="number" min="0" />
          </>
        )}
      </div>
    </div>
  );
}

export default InfoBox;
