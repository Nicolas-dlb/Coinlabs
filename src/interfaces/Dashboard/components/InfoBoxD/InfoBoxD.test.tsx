/* eslint-disable no-unused-vars */
import { mount } from "enzyme";
import React from "react";
import store from "redux/store";
import { Provider } from "react-redux";
import InfoBoxD from "./InfoBoxD";

describe("InfoBoxD", () => {
  it("should render correctly", () => {
    const InfoBoxDComponent = mount(
      <Provider store={store}>
        <InfoBoxD title="Balance" variation={26} />
      </Provider>
    );
    expect(InfoBoxDComponent).toMatchSnapshot();
  });
});
