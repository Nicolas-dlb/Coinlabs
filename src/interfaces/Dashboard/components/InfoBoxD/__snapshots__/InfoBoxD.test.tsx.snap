// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBoxD should render correctly 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <InfoBox
    title="Balance"
    variation={26}
  >
    <div
      className="infoboxD"
    >
      <div
        className="infobox_top"
      >
        <p>
          Balance
        </p>
        <div
          className="infobox_top_variation"
        >
          <svg
            fill="#37bb77"
            style={
              Object {
                "transform": "rotate(140deg)",
              }
            }
            viewBox="0 0 492 492"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M464.3 207.4l0.8 0.2H135.9l103.5-103.7c5.1-5.1 7.8-11.9 7.8-19.1 0-7.2-2.8-14-7.8-19.1L223.3 49.5c-5.1-5.1-11.8-7.9-19-7.9 -7.2 0-14 2.8-19 7.8L7.8 226.9C2.8 232 0 238.8 0 246c0 7.2 2.8 14 7.8 19.1l177.4 177.4c5.1 5.1 11.8 7.8 19 7.8 7.2 0 13.9-2.8 19-7.8l16.1-16.1c5.1-5.1 7.8-11.8 7.8-19 0-7.2-2.8-13.6-7.8-18.7L134.7 284.4h330c14.8 0 27.3-12.8 27.3-27.6v-22.8C492 219.2 479.2 207.4 464.3 207.4z"
              strokeWidth="2"
            />
          </svg>
          <p
            style={
              Object {
                "color": "#37bb77",
              }
            }
          >
            26
            %
          </p>
        </div>
      </div>
      <div
        className="infobox_bottom"
      >
        <p>
          $
        </p>
        <p>
          20 000.00
        </p>
      </div>
    </div>
  </InfoBox>
</Provider>
`;
